cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/JsonMenu/index.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import If from 'components/If';
+import ClosedJsonMenu from 'components/PluginJSONCreator/Create/Content/JsonMenu/ClosedJsonMenu';
+import JsonLiveViewer from 'components/PluginJSONCreator/Create/Content/JsonMenu/JsonLiveViewer';
+import {
+  CreateContext,
+  createContextConnect,
+  ICreateContext,
+} from 'components/PluginJSONCreator/CreateContextConnect';
+import * as React from 'react';
+
+const JsonMenuView: React.FC<ICreateContext> = ({
+  pluginName,
+  pluginType,
+  displayName,
+  emitAlerts,
+  emitErrors,
+  configurationGroups,
+  groupToInfo,
+  groupToWidgets,
+  widgetInfo,
+  jsonView,
+  setJsonView,
+}) => {
+  return (
+    <div>
+      <If condition={jsonView}>
+        <JsonLiveViewer
+          pluginName={pluginName}
+          pluginType={pluginType}
+          displayName={displayName}
+          emitAlerts={emitAlerts}
+          emitErrors={emitErrors}
+          configurationGroups={configurationGroups}
+          groupToInfo={groupToInfo}
+          groupToWidgets={groupToWidgets}
+          widgetInfo={widgetInfo}
+          jsonView={jsonView}
+          setJsonView={setJsonView}
+        />
+      </If>
+      <If condition={!jsonView}>
+        <ClosedJsonMenu
```

Comment:
Can the liveviewer and closed json menu component be combined? It seemed like the contents are very similar

Reply:
The PR #12310  addresses this issue with a lot of their content being similar. For all the contents that are being shared between liveviewer and closed json menu, it is shared at https://github.com/cdapio/cdap/blob/882eb810d3291ed6b39efaf6394a932584abb867/cdap-ui/app/cdap/components/PluginJSONCreator/Create/PluginJSONMenu/index.tsx#L234. This minimizes repeating code between two components, and now each of these components (https://github.com/cdapio/cdap/blob/882eb810d3291ed6b39efaf6394a932584abb867/cdap-ui/app/cdap/components/PluginJSONCreator/Create/PluginJSONMenu/LiveViewMenu/index.tsx, https://github.com/cdapio/cdap/blob/882eb810d3291ed6b39efaf6394a932584abb867/cdap-ui/app/cdap/components/PluginJSONCreator/Create/PluginJSONMenu/CollapsedMenu/index.tsx) do not share any similar contents.

Reply:
This minimizes repeating code between two components, and now each of these components (https://github.com/cdapio/cdap/blob/882eb810d3291ed6b39efaf6394a932584abb867/cdap-ui/app/cdap/components/PluginJSONCreator/Create/PluginJSONMenu/LiveViewMenu/index.tsx, https://github.com/cdapio/cdap/blob/882eb810d3291ed6b39efaf6394a932584abb867/cdap-ui/app/cdap/components/PluginJSONCreator/Create/PluginJSONMenu/CollapsedMenu/index.tsx) do not share any similar contents.